tests/components/devolo_home_network/test_binary_sensor.py

```diff
@@ -0,0 +1,127 @@
+    LONG_UPDATE_INTERVAL,
+)
+from homeassistant.const import (
+    ENTITY_CATEGORY_DIAGNOSTIC,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry
+from homeassistant.util import dt
+
+from . import configure_integration
+from .const import PLCNET_ATTACHED
+
+from tests.common import async_fire_time_changed
+
+
+@pytest.mark.usefixtures("mock_device", "mock_zeroconf")
+async def test_binary_sensor_setup(hass: HomeAssistant):
+    """Test default setup of the binary sensor component."""
+    entry = configure_integration(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(f"{DOMAIN}.{FIRMWARE_UPDATE_AVAILABLE}") is not None
+    assert hass.states.get(f"{DOMAIN}.{CONNECTED_TO_ROUTER}") is None
+
+    await hass.config_entries.async_unload(entry.entry_id)
+
+
+@pytest.mark.usefixtures("mock_device", "mock_zeroconf")
+async def test_update_firmware_update_available(hass: HomeAssistant):
+    """Test state change of a firmware_update_available binary sensor device."""
+    state_key = f"{DOMAIN}.{FIRMWARE_UPDATE_AVAILABLE}"
+
+    entry = configure_integration(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+    assert state.state == STATE_OFF
+
+    er = entity_registry.async_get(hass)
+    assert er.async_get(state_key).entity_category == ENTITY_CATEGORY_DIAGNOSTIC
+
+    # Emulate device failure
+    with patch(
+        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_check_firmware_available",
+        side_effect=DeviceUnavailable,
+    ):
+        async_fire_time_changed(hass, dt.utcnow() + LONG_UPDATE_INTERVAL)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(state_key)
+        assert state is not None
+        assert state.state == STATE_UNAVAILABLE
+
+    # Emulate state change
+    with patch(
+        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_check_firmware_available",
+        new=AsyncMock(return_value={"result": "UPDATE_AVAILABLE"}),
+    ):
+        async_fire_time_changed(hass, dt.utcnow() + LONG_UPDATE_INTERVAL)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(state_key)
+        assert state is not None
+        assert state.state == STATE_ON
+
+    await hass.config_entries.async_unload(entry.entry_id)
+
+
+@pytest.mark.usefixtures("mock_device", "mock_zeroconf")
+async def test_update_attached_to_router(hass: HomeAssistant):
+    """Test state change of a attached_to_router binary sensor device."""
+    state_key = f"{DOMAIN}.{CONNECTED_TO_ROUTER}"
+    entry = configure_integration(hass)
+    with patch(
+        "homeassistant.helpers.entity.Entity.entity_registry_enabled_default",
+        return_value=True,
+    ):
```

Comment:
Instead of patching home assistant, update the entity in the registry to enable it

Reply:
By spying at other integrations, there seem to be two ways to do that. I chose the one that seemed simpler to me.